sdk/communication/azure-communication-chat/tests.yml

```diff
@@ -19,5 +19,13 @@ stages:
         groupId: com.azure
         safeName: azurecommunicationchat
     ServiceDirectory: communication
+    PostSteps:
+      - task: PublishCodeCoverageResults@1
```

Comment:
We should try to avoid duplicating these steps everywhere. In fact we probably should avoid duplicating a set of these steps so by creating a template you can share across your packages.

Reply:
Let me try

Reply:
Made a few attempts. Sorry, they didn't work out. It is not obvious to me how this can be done without modifying code in eng\pipelines folder. If adding a new variable can be a concern, I am afraid I shouldn't be making big changes there for the purpose of this particular PR. @weshaggard

Reply:
I've not tested these but this is what I was thinking:

Each individual template would look like
```
trigger: none

stages:
- template: /sdk/communication/tests-template.yml
  parameters:
    PackageName: azure-communication-chat
    SafeName: azurecommunicationchat
```
Add a new tests-template.yml file in the communication directory that looks something like:
```parameters:
  PackageName: not-specified
  SafeName: notspecified

stages:
- template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
  parameters:
    CloudConfig:
      Public:
        SubscriptionConfigurations:
          - $(sub-config-azure-cloud-test-resources)
          - $(sub-config-communication-services-cloud-test-resources-common)
          - $(sub-config-communication-services-cloud-test-resources-java)
      Int:
        SubscriptionConfigurations:
          - $(sub-config-communication-int-test-resources-common)
          - $(sub-config-communication-int-test-resources-java)
    Clouds: Public, Int
    Artifacts:
      - name: ${{ parameters.PackageName }}
        groupId: com.azure
        safeName: ${{ parameters.SafeName }}
    ServiceDirectory: communication
    PostSteps:
      - task: PublishCodeCoverageResults@1
        condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['JavaTestVersion'], '1.11'), eq(variables['CurrentCloud'], 'Public'))
        inputs:
          codeCoverageTool: JaCoCo
          summaryFileLocation: sdk/communication/${{ parameters.PackageName }}/target/site/jacoco/jacoco.xml
          reportDirectory: sdk/communication/${{ parameters.PackageName }}/target/site/jacoco/
          failIfCoverageEmpty: true
    EnvVars:
      SKIP_LIVE_TEST: TRUE
```

That would eliminate a lot of duplication and make sure things remain consistent across packages.

Reply:
I think I tried that and it complained about parameters declaration. Let me try again.

Reply:
@weshaggard That worked! Thank you! Updated. Please take another look

Reply:
Yes that looks like what I was looking for and helps reduce lot of duplication and increases consistency.